Action can wait for at least 27 minutes when no progress is being made on the download

Example: https://github.com/ccmbioinfo/osmp/runs/6692774526?check_suite_focus=true

This hung on `Received 1189052508 of 1193246812 (99.6%)` for almost half an hour before we noticed and cancelled the workflow run.

hmmm I'm seeing a situation where `SEGMENT_DOWNLOAD_TIMEOUT_MIN` seems to not have worked

here's the workflow run: https://github.com/getsentry/sentry/actions/runs/3113974617/jobs/5049272594

it's a little obscured but the environment variable is set at the workflow level here: https://github.com/getsentry/sentry/blob/1ba48063464c06aea8f0a02b7fe797bfb6cab715/.github/workflows/frontend.yml#L15-L17

then a composite action is called here: https://github.com/getsentry/sentry/blob/1ba48063464c06aea8f0a02b7fe797bfb6cab715/.github/workflows/frontend.yml#L146

and then that composite action calls actions/cache here: https://github.com/getsentry/sentry/blob/1ba48063464c06aea8f0a02b7fe797bfb6cab715/.github/actions/setup-volta/action.yml#L13

the output of the job with the relevant section:

```
2022-09-23T15:50:17.5651366Z ##[group]Run actions/cache@56046cbc4743437ac40542086317b1561d7705f8
2022-09-23T15:50:17.5651651Z with:
2022-09-23T15:50:17.5651888Z   path: /home/runner/.cache/yarn/v6
2022-09-23T15:50:17.5652379Z   key: linux-volta-1.0.8-node-16.13.1-yarn-1.22.5-64094b67dee5167fb551a5fbe8757dd24aa02132dea9048f118c4c4d2f5ea311
2022-09-23T15:50:17.5652805Z env:
2022-09-23T15:50:17.5653046Z   SEGMENT_DOWNLOAD_TIMEOUT_MIN: 3
2022-09-23T15:50:17.5653322Z   VOLTA_HOME: /home/runner/.volta
2022-09-23T15:50:17.5653576Z ##[endgroup]
2022-09-23T15:50:19.2224966Z Received 0 of 180889536 (0.0%), 0.0 MBs/sec
2022-09-23T15:50:20.2230569Z Received 83886080 of 180889536 (46.4%), 39.9 MBs/sec
2022-09-23T15:50:21.2226308Z Received 176695232 of 180889536 (97.7%), 56.1 MBs/sec
2022-09-23T15:50:22.2239739Z Received 176695232 of 180889536 (97.7%), 42.1 MBs/sec
2022-09-23T15:50:23.2253786Z Received 176695232 of 180889536 (97.7%), 33.7 MBs/sec
2022-09-23T15:50:24.2268647Z Received 176695232 of 180889536 (97.7%), 28.1 MBs/sec
2022-09-23T15:50:25.2280039Z Received 176695232 of 180889536 (97.7%), 24.0 MBs/sec
2022-09-23T15:50:26.2289787Z Received 176695232 of 180889536 (97.7%), 21.0 MBs/sec
2022-09-23T15:50:27.2297324Z Received 176695232 of 180889536 (97.7%), 18.7 MBs/sec
2022-09-23T15:50:28.2309270Z Received 176695232 of 180889536 (97.7%), 16.8 MBs/sec
2022-09-23T15:50:29.2313534Z Received 176695232 of 180889536 (97.7%), 15.3 MBs/sec
2022-09-23T15:50:30.2327106Z Received 176695232 of 180889536 (97.7%), 14.0 MBs/sec
2022-09-23T15:50:31.2340917Z Received 176695232 of 180889536 (97.7%), 12.9 MBs/sec
2022-09-23T15:50:32.2356730Z Received 176695232 of 180889536 (97.7%), 12.0 MBs/sec
2022-09-23T15:50:33.2368999Z Received 176695232 of 180889536 (97.7%), 11.2 MBs/sec
2022-09-23T15:50:34.2373159Z Received 176695232 of 180889536 (97.7%), 10.5 MBs/sec
2022-09-23T15:50:35.2386669Z Received 176695232 of 180889536 (97.7%), 9.9 MBs/sec
...

2022-09-23T16:41:49.0253357Z Received 176695232 of 180889536 (97.7%), 0.1 MBs/sec
2022-09-23T16:41:50.0265697Z Received 176695232 of 180889536 (97.7%), 0.1 MBs/sec
2022-09-23T16:41:51.0279084Z Received 176695232 of 180889536 (97.7%), 0.1 MBs/sec
2022-09-23T16:41:52.0291969Z Received 176695232 of 180889536 (97.7%), 0.1 MBs/sec
2022-09-23T16:41:53.0077523Z ##[error]The operation was canceled.
```

I expect my setting of the variable to 3 to have it time out at 3 minutes

--- a/.github/workflows/build_and_test_and_release.yml
+++ b/.github/workflows/build_and_test_and_release.yml
@@ -24,7 +24,7 @@ jobs:
         uses: actions/setup-python@v2
         with:
           python-version: "3.8"
-      - uses: actions/cache@v1
+      - uses: actions/cache@v3
         with:
           path: ~/.cache/pip
           key: pip_cache
@@ -38,7 +38,7 @@ jobs:
         run: "./.ci/install_prereqs.sh"
       - name: "Install package"
         run: ./.ci/install.sh
-      - uses: actions/cache@v1
+      - uses: actions/cache@v3
         with:
           path: ~/docker_images
           key: docker_cache

